File: ui/src/synergy/SynTraitHouse.ts
import { SynTraitHouse } from "../generated-src/SynTraitHouse"

export const synTraitHouseShortLabel = (house: SynTraitHouse) => {
    switch (house) {
        case SynTraitHouse.anyHouse:
            return "Any"
        case SynTraitHouse.house:
            return "House"
        case SynTraitHouse.outOfHouse:
            return "Out of"
        case SynTraitHouse.continuous:
            return "Omni"
    }
}

File: src/main/kotlin/coraythan/keyswap/synergy/SynTraitHouse.kt
package coraythan.keyswap.synergy

import coraythan.keyswap.generatets.GenerateTs

@GenerateTs
enum class SynTraitHouse {
    anyHouse,
    // Only synergizes with traits inside its house
    house,
    outOfHouse,
    continuous,
}
